src/App.js

```diff
@@ -1,60 +1,56 @@
-const DB_MESSAGES_KEY = "messages";
+export const App = () => {
+  const [loggedInUser, setLoggedInUser] = useState(false);
+  const [user, setUser] = useState({});
 
-class App extends React.Component {
-  constructor(props) {
-    super(props);
-    // Initialise empty messages array in state to keep local state in sync with Firebase
-    // When Firebase changes, update local state, which will update local UI
-    this.state = {
-      messages: [],
-    };
-  }
+  const signOutUser = () => {
+    setLoggedInUser(false);
+    setUser({});
+    signOut(auth);
+  };
 
-  componentDidMount() {
-    const messagesRef = ref(database, DB_MESSAGES_KEY);
-    // onChildAdded will return data for every child at the reference and every subsequent new child
-    onChildAdded(messagesRef, (data) => {
-      // Add the subsequent child to local component state, initialising a new array to trigger re-render
-      this.setState((state) => ({
-        // Store message key so we can use it as a key in our list items when rendering messages
-        messages: [...state.messages, { key: data.key, val: data.val() }],
-      }));
+  useEffect(() => {
+    onAuthStateChanged(auth, (user) => {
+      if (user) {
+        setLoggedInUser(true);
+        setUser(user);
+      }
     });
-  }
-
-  // Note use of array fields syntax to avoid having to manually bind this method to the class
-  writeData = () => {
-    const messageListRef = ref(database, DB_MESSAGES_KEY);
-    const newMessageRef = push(messageListRef);
-    set(newMessageRef, "abc");
-  };
+  }, []);
 
-  render() {
-    // Convert messages in state to message JSX elements to render
-    let messageListItems = this.state.messages.map((message) => (
-      <li key={message.key}>{message.val}</li>
-    ));
-    return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <p>
-            Edit <code>src/App.js</code> and save to reload.
+  return (
+    <>
+      <div className="flex flex-col justify-between lg:flex-row ">
+        {loggedInUser ? (
+          <p className="btn btn-ghost text-xl">Hello, {user.email}</p>
+        ) : (
+          <p className="btn btn-ghost text-xl">
+            Please sign in to chat and post~
           </p>
-          {/* TODO: Add input field and add text input as messages in Firebase */}
-          <button onClick={this.writeData}>Send</button>
-          <ol>{messageListItems}</ol>
-        </header>
+        )}
+        <NavBar loggedInUser={loggedInUser} signOut={signOutUser} />
       </div>
-    );
-  }
-}
-
-export default App;
+      <div className="flex flex-col items-start pb-10 lg:flex-row lg:justify-around">
+        <div className="flex flex-col h-[680px] mt-2">
+          <ChatMessagesFunction />
+          <div className="flex lg:justify-between m-2">
+            {loggedInUser ? <MessageFormFunction email={user.email} /> : null}
```

Comment:
```suggestion
            {loggedInUser && <MessageFormFunction email={user.email} />}
```

Ideally include the div also in that conditional rendering